src/main.cpp

```diff
@@ -264,4 +273,86 @@ void runSteeringDisplay() {
 	while (1) {
 		display.run();
 	}
+}
+
+void buttonRead(int buttonPress){
+	int ledPos;
+	for(int i = 0; i <8; i++){
+		shiftClk = 1;
+		wait_us(20);
+		shiftClk = 0;
+		shiftLatch = 1;
+
+		if(i == 3 && buttonPress == 1){
+			ledPos = 1;
+			ledOn(ledPos);
+		}
+
+		else if(i == 4 && buttonPress == 1){
+			ledPos = 2;
+			ledOn(ledPos);
+		}
+
+		else if(i == 5 && buttonPress == 1){
+			ledPos = 3;
+			ledOn(ledPos);
+		}
+
+		else if(i == 6 && buttonPress == 0){
+			ledPos = 4;
+			ledOn(ledPos);
+		}
+
+		else if(i == 6 && buttonPress == 1){
+			ledPos = 5;
+			ledOn(ledPos);
+		}
+
+		else if(i == 7 && buttonPress == 1){
+			ledPos = 6;
+			ledOn(ledPos);
+		}
+
+		wait_us(20);
+		shiftLatch = 0;
+	}
+}
+
+void ledOn(int ledPos){
+	for(int i = 0; i < 8; i++){
+		shiftClk = 1;
+		wait_us(20);
+		shiftClk = 0;
+		shiftLatch = 1;
+		wait_us(20);
+		shiftLatch = 0;
+
+		if(i == ledPos) ledOut = 1;
+	}
+}
+
+void buttonTest(int buttonPress){
+	shiftClk = 1;
+	shiftLatch = 0;
+	if(buttonPress == 1) ledOut.write(1);
+}
+
+void buttonState(){
+	shiftClk = 0;
+
+	buttons = (buttons << 1) | buttonIn.read();
```

Comment:
It looks like this shifts in the current button read to the least-significant bit, cool idea

Reply:
Yep, the "buttons" variable will hold the current state of all the buttons